controllers/ingressController.go

```diff
@@ -72,22 +66,17 @@ func WhiteListIP(c *gin.Context) {
 		return
 	}
 	leases = GetActiveLeases(ns, name)
-	for kubeContext, kubeClient := range config.KubeClients {
-		clientset := kubeClient.ClientSet
-		myclientset := pkg.MyClientSet{Clientset: clientset}
-		data, err = myclientset.GetIngress(kubeContext, ns, name)
-		if err != nil {
-			errs = errs + 1
-		}
-		if data.Name != "" {
-			myIngress = data
-			break
-		}
+
+	showIngressDetailsRequest := ingress_driver.ShowIngressDetailsRequest{
+		Namespace: ns,
+		Name:      name,
 	}
 
-	if errs >= len(config.KubeClients) {
+	showIngressDetailsResponse, err := ingress_driver.GetIngressDriverForNamespace(ns).ShowIngressDetails(showIngressDetailsRequest)
```

Comment:
Is `GetIngressDriverForNamespace()` really needed if we are already passing the namespace in `ShowIngressDetails()`

Reply:
changed to avoid passing to ShowIngressDetails. GetIngressDriverForNamespace is needed to select the driver